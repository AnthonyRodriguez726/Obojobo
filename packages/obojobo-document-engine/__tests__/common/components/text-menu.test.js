import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import TextMenu from '../../../src/scripts/common/components/text-menu'

let createRect = () => {
	return {
		left: 0,
		top: 0,
		width: 0
	}
}
describe('TextMenu', () => {
	test('TextMenu', () => {
		const component = renderer.create(<TextMenu />)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('TextMenu disabled', () => {
		const component = renderer.create(<TextMenu relativeToElement={true} />)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('TextMenu with no selection', () => {
		const component = renderer.create(
			<TextMenu
				relativeToElement={{
					getBoundingClientRect: () => createRect()
				}}
				enabled={true}
				selectionRect={createRect()}
			/>
		)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('TextMenu with required props', () => {
		const component = renderer.create(
			<TextMenu
				relativeToElement={{
					getBoundingClientRect: () => createRect()
				}}
				enabled={true}
				selectionRect={createRect()}
				commands={[{ label: 'Label' }, { label: 'imageCommand', image: 'mockImage' }]}
			/>
		)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('TextMenu calls mouseDown', () => {
		let mockCommandHandler = jest.fn()

		const component = mount(
			<TextMenu
				relativeToElement={{
					getBoundingClientRect: () => createRect()
				}}
				enabled={true}
				selectionRect={createRect()}
				commands={[{ label: 'Label' }]}
				commandHandler={mockCommandHandler}
			/>
		)

		let link = component.find('a').simulate('mouseDown')

		expect(mockCommandHandler).toHaveBeenCalledWith('Label')
	})
})
