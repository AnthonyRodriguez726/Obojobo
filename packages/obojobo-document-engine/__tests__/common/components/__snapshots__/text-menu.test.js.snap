// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextMenu 1`] = `null`;

exports[`TextMenu with required props 1`] = `
<div
  className="editor--components--text-menu"
  style={
    Object {
      "left": "0px",
      "top": "0px",
    }
  }
>
  <a
    onMouseDown={[Function]}
  >
    <div>
      <span>
        Label
      </span>
      <img
        className="click-blocker"
      />
    </div>
  </a>
</div>
`;
